### DISCUSSION TITLE
How to set up Blog as home page?

### DISCUSSION THREAD

**Discussion by s-c-a-l-l-o-p** [thuliteio/doks#1255] [Q&A]
I tried to copy the code from:

node_modules/@hyas/doks-core/layouts/blog/list.html <--- THE LIST OF ALL BLOG POST TEMPLATE

to

layouts/index.html <--- THE ACTUAL DOKS HOME PAGE

in order to have the Home page listing all my posts, but the result was this:

```
render: failed to render pages: render of "home" failed: "/home/scallop/the/furry/shameless/fest/layouts/index.html:43:15": execute of template failed at <partial "main/blog-meta.html" .>: error calling partial: "/home/scallop/the/furry/shameless/fest/layouts/node_modules/@hyas/doks-core/layouts/partials/main/blog-meta.html:1:17": execute of template failed at <len .Params.contributors>: error calling len: reflect: call of reflect.Value.Type on zero Value
```
And this error message here too:

```
/home/scallop/the/furry/shameless/fest/node_modules/@hyas/doks-core/layouts/partials/main/blog-meta.html:1:17:
```

```
{{ $last := sub (len .Params.contributors) 1 -}}
<p>
  <small>
```

So a I got back to layouts/index.html and removed this line of code:

```
        <div class="card-body">
          <article>
            <h2 class="h3"><a class="stretched-link text-body" href="{{ .RelPermalink }}">{{ .Params.title }}</a></h2>
            <p>{{ .Params.summary | safeHTML }}</p>
            {{ partial "main/blog-meta.html" . -}} <----THIS LINE RIGHT HERE
          </article>
        </div>
``` 

After removing that line, Hugo started working again and was not angry with me anymore. 🥲 

This is how my Home page looks like after that change:

![doks_print_69](https://github.com/gethyas/doks/assets/16630637/127543c5-d994-47b4-83a2-ffa26e0740bc)

Here is the original blog list from Doks:

![doks_69_print](https://github.com/gethyas/doks/assets/16630637/390ebb1d-c8c7-41e6-ad06-03045376f1c6)

The distance from top menu changed and all the values they got removed because I was forced to remove that blog-meta.html file.

I read the entire documentation and all discussion here and I didn't found any solution, not sure if there's one. 

Any help will be really appreciated and your mom, dad and the president of your country will love you more.  🫶

**Comment by h-enk** [comment]
What about adding to the frontmatter in `content/blog/_index.md`:

```md
url: "/"
```

For more background, see the Hugo docs: [`url`](https://gohugo.io/content-management/urls/#url)

**Reply by s-c-a-l-l-o-p** [reply]
> What about adding to the frontmatter in `content/blog/_index.md`:
> 
> ```
> url: "/"
> ```

That doesn't work because **/layouts/index.html** exist, so Hugo will pick that layout as Home page.

If **/layouts/index.html** is renamed or deleted, Hugo will pick **node_modules/@hyas/doks-core/layouts/index.html** as the Home page.

So the only solution here is this:

1 - Go to **content/blog/_index.md** as @h-enk suggested and add **url: "/"** in frontmatter.

2 - Go to **/layouts/index.html** and remove all the code, yes remove all, don't delete the file.

So now, Hugo will find the **/layouts/index.html** but it'll be empty, and the Blog frontmatter will tell Hugo that this is the Home page. 
Somehow this works to redefine the Home page. 

If feels kind of *hackish*, but it worked so far.

The ideal solution here is to re-write the **/layouts/index.html** template to list all posts from Blog. I'll do that later, but for now, I'll work with this solution because I'm running out of time.

Thank you @h-enk for the insight and amazing Hugo theme.

**Reply by h-enk** [reply]
Thanks and you're welcome.

> The ideal solution here is to re-write the /layouts/index.html template to list all posts from Blog. 

~~That's right, I agree with that.~~ It's one of the options. I recommend to use this one: https://github.com/gethyas/doks/discussions/1255#discussioncomment-10312520

**Reply by satrox28** [reply]
Hi @h-enk  I am looking for the same, apart from removing the content from ` /layouts/index.html ` can users fork the theme and modify the redirect link from home page accordingly?.

**Reply by h-enk** [reply]
There's no need to remove the content from `/layouts/index.html`!

Just add to the frontmatter in `content/blog/_index.md`:

```md
url: "/"
```

See also:
- [Example repo](https://github.com/gethyas/doks-1255)
- [Example website](https://marvelous-valkyrie-ff5f28.netlify.app/)

**Reply by satrox28** [reply]
Thank You, will take a look.
